### ISSUE TITLE
Server-side slug generation should respect WAGTAIL_ALLOW_UNICODE_SLUGS 

### ISSUE THREAD

**Issue by gasman** [wagtail/wagtail#5805]
Currently the `WAGTAIL_ALLOW_UNICODE_SLUGS` setting only affects the behaviour of the client-side slug generation on the page creation/edit view. This means that when pages are created programmatically, the slug can still contain non-ASCII characters:

```
>>> from wagtail.core.models import Page
>>> page = Page(title="A mööse bit me önce")
>>> Page.get_first_root_node().add_child(instance=page)
<Page: A mööse bit me önce>
>>> page.slug
'a-mööse-bit-me-önce'
```

In this case, the slug generation happens in `Page.full_clean`, which currently calls `slugify` with `allow_unicode=True` - this should pass the value of `WAGTAIL_ALLOW_UNICODE_SLUGS` instead.
https://github.com/wagtail/wagtail/blob/afd707f6b40e63faaccbd743bdc0fc2111af0043/wagtail/core/models.py#L434

**Comment by ArkadiuszMichalRys** [comment]
I would like to try fix this one.

**Comment by gasman** [comment]
Fixed in #5808
